### PR TITLE
Select wildcard functionality implemented

### LINKED ISSUE #1432: Wire up select * query
We should be able to do a `SELECT * from cpu` type of query.

However, we don't need to support * as an argument to aggregate functions.

### PR DESCRIPTION
**Pull request by corylanou**
Fixes #1432

### CODE REVIEW

**Comment by otoolep** [comment]
I understand how this works, but shouldn't it be performed on the *sub-statements*? For example, this won't work, if I understand our system correctly:

`SELECT *,value from cpu`

A pretty stupid query, I know, but will this blow things up? @benbjohnson

**Comment by corylanou** [comment]
It won't currently blow up, but it's not currently supported either.  First pass was to support a single wildcard.  We purposefully put in `if len(stmt.Fields) == 1 {` just to check for that.  I just tested locally and it returns no results.

It should probably return an error if it sees a wildcard and another field.

File: server_test.go [hunk 1]
```diff
@@ -799,6 +799,10 @@ func TestServer_ExecuteQuery(t *testing.T) {
 	s.MustWriteSeries("foo", "raw", []influxdb.Point{{Name: "cpu", Tags: map[string]string{"region": "us-east"}, Timestamp: mustParseTime("2000-01-01T00:00:10Z"), Values: map[string]interface{}{"value": float64(30)}}})
 	s.MustWriteSeries("foo", "raw", []influxdb.Point{{Name: "cpu", Tags: map[string]string{"region": "us-west"}, Timestamp: mustParseTime("2000-01-01T00:00:00Z"), Values: map[string]interface{}{"value": float64(100)}}})
 
+	// TODO corylanou: make this test work after it waits enough time to flush out writes.
+	// It is currently passing by accident, as the last write doesn't have enough time to show up and the tests below should not pass
+	// time.Sleep(1 * time.Second)
```
Comment (otoolep) [inline server_test.go]: Did you mean to remove this code completely?

**Comment by otoolep** [comment]
Since the point of this code is to ensure that `SELECT *` works as expected, we should have a dedicated test that writes more than 1 field for a datapoint, and then ensures that all the expected fields are returned in response to `SELECT *`.

In other words I am not sure if we're really testing this new feature - let me know if I am missing something.

**Comment by corylanou** [comment]
@otoolep good point on the tests.  I created a test that covers multiple values in the write and query.  Let me know what you think.

**Comment by otoolep** [comment]
Much better test @corylanou -- thanks.

File: server.go [hunk 2]
```diff
@@ -1848,10 +1848,36 @@ func (s *Server) planSelectStatement(stmt *influxql.SelectStatement, database st
...
+	isWildcard := false
+	for _, f := range stmt.Fields {
+		if _, ok := f.Expr.(*influxql.Wildcard); ok {
+			isWildcard = true
+			break
+		}
+	}
+
+	if len(stmt.Fields) != 1 && isWildcard {
+		return nil, fmt.Errorf("unsupported query: %s.  currently only single wildcard is supported.", stmt.String())
```
Comment (otoolep) [inline server.go]: Nit: This sounds like `SELECT *,value` is supported, since it's still a single wildcard. :-) "....only solitary wildcard supported."?

**Comment by otoolep** [comment]
Functionality looks good. +1 modulo future fix-ups.

**Comment by corylanou** [comment]
@otoolep as an FYI, the query parser actually errors out on the `select *,value from "foo"."bar".cpu` before it gets to the code I added with the following error:
`error parsing query: found ,, expected FROM at line 1, char 9`

So we don't support it, and we return an intelligent error to the user as well.

**Comment by corylanou** [comment]
@otoolep let me know if this is good to merge.

**Comment by otoolep** [comment]
Looks fine, I suggest we open a ticket so we return to this with a more coherent fix sometime.

**Pull request merged**